boot/bootutil/src/loader.c

```diff
@@ -965,7 +983,8 @@ boot_swap_image(struct boot_loader_state *state, struct boot_status *bs)
 }
 #endif
 
-#if (BOOT_IMAGE_NUMBER > 1)
+#if (BOOT_IMAGE_NUMBER > 1) || \
+    (defined(MCUBOOT_OVERWRITE_ONLY) && defined(MCUBOOT_DOWNGRADE_PREVENTION))
```

Comment:
The definition will need to be moved earlier in the file. Currently, this causes a compile fail in the simulator (and probably at least warnings on any target). It needs to be moved earlier in the file that its use above.

Reply:
Fixed